src/AzSK.ADO/Framework/BugLog/AutoCloseBugManager.ps1

```diff
@@ -90,18 +95,38 @@ class AutoCloseBugManager {
 
                 if ($this.UseAzureStorageAccount -and $this.ScanSource -eq "CA")
                 {
+                    $tagHash=$this.GetHashedTag($control.ControlItem.Id, $control.ResourceContext.ResourceId)
+                    $hashToControlIDMap.add($tagHash,$control);
                     #complete the query
-                    $TagSearchKeyword += "(ADOScannerHashId eq '" + $this.GetHashedTag($control.ControlItem.Id, $control.ResourceContext.ResourceId) + "') or "
+                    $TagSearchKeyword += "(ADOScannerHashId eq '" + $tagHash + "') or "
                     #if the query count equals the passing control results, search for bugs for this batch
                     if ($QueryKeyWordCount -eq $PassedControlResultsLength) {
                         #to remove OR from the last tag keyword. Ex: Tags: Tag1 OR Tags: Tag2 OR. Remove the last OR from this keyword
                         $TagSearchKeyword = $TagSearchKeyword.Substring(0, $TagSearchKeyword.length - 3)
-                        $response = $this.BugLogHelperObj.GetTableEntityAndCloseBug($TagSearchKeyword)
+                        $closedBugsResponse = $this.BugLogHelperObj.GetTableEntityAndCloseBug($TagSearchKeyword)
+                        if ($closedBugsResponse){
```

Comment:
keep this in a common method and call that method

Reply:
Added a common method